qualtran/surface_code/physical_parameters.py

```diff
@@ -0,0 +1,50 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from attrs import field, frozen
+
+
+@frozen
+class PhysicalParameters:
+    """The physical properties of a quantum computer..
+
+    Attributes:
+        t_gate_ns: Clifford gate time.
+        t_meas_ns: Measurement time.
+        physical_error_rate: Physical error rate.
+        reference: Source of these estimates.
+    """
+
+    t_gate_ns = field(default=1, repr=lambda x: f'{x:g}')
+    t_meas_ns = field(default=1, repr=lambda x: f'{x:g}')
+
+    physical_error_rate = field(default=1e-3, repr=lambda x: f'{x:g}')
+
+    reference = field(default='')
+
+
+FowlerGidney = PhysicalParameters(
+    t_gate_ns=100,
+    t_meas_ns=100,
```

Comment:
where in the reference did you get these values? in your original PR you had these set to 1us

Reply:
they were wrong in the first PR. 

The latencies in https://arxiv.org/pdf/1808.06709.pdf are not specifically mentioned, but in the conclusion section it says `surface code error detection circuit time of 1µs`. For a gate based surface code this would be $(4 t_{gate} + 2 t_{meas}) d$. So I just estimate them by ssuming `t_{gate} = t_{meas}` and using $d=7$ to be ~23ns.

but these are all estimate so they are all wrong in a sense

Reply:
1. please document that in the code. We want to reduce the quantity of unexplained magic numbers; not increase them!
2. The 1us is independent of d. Doing d cycles would be d*1us. This is important to get correct! 

We should never be assuming a particular value for `d`. That is an important design parameter that the application-executor gets to decide

Reply:
I didn't mean that I'm assuming `d = 7` for all use case. I mean that in the paper they sometimes use $d = 7$ and sometimes used $d = 15$ and it's not clear to me which one they used when they calculated $1\mu s$.

Reply:
the 1us number is independent of d. It's the (assumed) time it takes to run four layers of cnots and one layer of measurement; i.e. one cycle of error correction; gates are tens of nanoseconds and measurement is more-or-less 1us

Reply:
Where did you get the formula (4 tgate + 2 tmeas)*d ?

Reply:
table 5 of https://arxiv.org/pdf/2211.07629.pdf for gate based surface code

Reply:
Got it.

Do you know why it's 2*t_meas? I thought each cycle included one moment of measurement.
 
Please note that that formula is labeled "logical time step". A lot of terms in these costs models are overloaded. That's why it's so important to get standardized terminology in code with everything documented (i.e. this module). There are n relevant times
 - the smallest division is like `cirq.Moment` and the time it takes to execute a (set of parallel) gate(s). On google's architecture: for CNOT this is ~tens of nanoseconds. For Measurement this is hundreds of  nanoseconds
 - A surface code cycle (aka round) is the time it takes to measure the stabilizers once. This is approx 4t_gate + t_meas. In the gidney/fowler models this is the smallest unit of time considered. They usually assume 1us because that's a nice round number that is close to 4tgate+tmeas.
 - A "logical time step" involves doing `d` cycles of error correction to pump out timelike errors.